### PR TITLE
Multiple changes

### PR DESCRIPTION
**Pull request by gomoripeti**

### CODE REVIEW

File: lib/gradualixir.ex [hunk 1]
```diff
@@ -4,7 +4,7 @@ defmodule Gradualixir do
   """
 
   @doc """
-  Hello world.
+  Examples speek for themselves
```
Comment (OvermindDL1) [inline lib/gradualixir.ex]: `s/speek/speak/`?  :-)

**Review by OvermindDL1 — CHANGES_REQUESTED** [review]
Looks great!  Just one minor change unless it's a pun I'm missing?  ^.^;

File: lib/gradualixir.ex (hunk 1 above)
Comment (gomoripeti) [reply lib/gradualixir.ex]: no pun, this is a real plain typo

**Comment by OvermindDL1** [comment]
Looking great!  Any other changes or tests or so you want to add in or do you consider it good to merge?  :-)

**Comment by gomoripeti** [comment]
that's all from me for now. I guess next time to bump gradualizer hash when the beta or first hex package comes out.
(I started to work on an Elixir formater earlier this year, but it's far from usable yet, unfortunately)

**Comment by OvermindDL1** [comment]
> (I started to work on an Elixir formater earlier this year, but it's far from usable yet, unfortunately)

It's easy enough to just call the built-in formatter, or PR changes to freedom_formatter, which is the formatter as a standalone project and some additional options.  :-)

**Comment by gomoripeti** [comment]
The challenge is that Gradualizer works on the Erlang AST. Reconstructing types is mostly possible because they are mostly identically and eg Erlex can help here (except for the Gradualizer specific extensions). But reconstructing any expression from Erlang AST to Elixir quoted expr is probably impossible. (Eg was it a `with` expression or embedded `case` expressions) Maybe the way forward is not to print the whole problematic expressions just point to the line number.

**Comment by OvermindDL1** [comment]
Elixir projects by default include the debug chunk in the `*.beam` files so you can pull the original expression out of that (Elixir really doesn't like the debug chunk being removed as it uses it for a lot of error handling).  Types are identical though so that part is fine, just some reformatting and such of them.

**Pull request merged**